qr-code-auth/README.md

```diff
@@ -0,0 +1,275 @@
+      }
+      ```
+
+- **Example:**
+  GET /auth/qr-code-auth?user_id=SooJK37gzjIZfFNH0tlL<br/>
+  Status: 200 OK<br/>
+  ```json
+  {
+    "message": "Authentication document retrieved successfully.",
+    "data": {
+      "user_id": "SooJK37gzjIZfFNH0tlL",
+      "device_info": "t5k77PHnuDSrgEzvMJAj",
+      "authorization_status": "NOT_INIT",
+      "access_token": ""
+    }
+  }
+  ```
+  GET /auth/qr-code-auth?user_id=invalidUserId<br/>
+  Status: 404 Not Found<br/>
+  ```json
+  {
+    "message": "No Authentication authentication found."
+  }
+  ```
+  GET /auth/qr-code-auth?taskId=GTB4UUtlKwGemRN2lwBp11<br/>
+  Status: 400 Bad Request
+  ```json
+  {
+    "statusCode": 400,
+    "error": "Bad Request",
+    "message": "invalid query parameters passed"
+  }
+  ```
+  GET /?user_id=GTB4UUtlKwGemRN2lwBp11
+  Status: 500 Internal Server Error
+  ```json
+  {
+    "message": "The server has encountered an unexpected error. Please contact the administrator for more information."
+  }
+  ```
+
+## PATCH /auth/qr-code-auth/authorization_status/{authorization_status}
+
+Updates the authorization_status field of an existing qr-code-auth document for the specified user.
+
+- **Params**
+  - **authorization_status** (required, string): Specifies the authorization status of the user. The possible values are "NOT_INIT" , "AUTHORIZED" , "REJECTED". The default value will be "NOT_INIT"
+- **Query**
+  None
+- **Headers**
+  Content-Type: application/json
+- **Cookie**  
+  rds-session: `<JWT>`
+- **Success Response:**
+  - **Code:** 200
+    - **Content:**
+      ```json
+      {
+        "message": "String",
+        "data": {
+          "user_id": "String",
+          "device_info": "String",
+          "authorization_status": "String",
+          "access_token": "String"
+        }
+      }
+      ```
+- **Error Response:**
+
+  - **Code:** 400
+
+    - **Content:**
+      ```json
+      {
+        "message": "Bad Request"
+      }
+      ```
+
+  - **Code:** 401
+    - **Content:**
+      ```json
+      {
+        "statusCode": 401,
+        "error": "Unauthorized",
+        "message": "Unauthenticated User."
+      }
+      ```
+  - **Code:** 404
+    - **Content:**
+      ```json
+      {
+        "message": "Document not found"
+      }
+      ```
+  - **Code:** 500
+    - **Content:**
+      ```json
+      {
+        "message": "The server has encountered an unexpected error. Please contact the administrator for more information."
+      }
+      ```
+
+- **Example for updating an existing authentication document:**
+  PATCH /auth/qr-code-auth/authorization_status/AUTHORIZED<br/>
+  Content-Type: application/json<br/>
+  Request-Body:<br/>
+  None
+  Response :
+  Status 200 OK <br/>
+  Content-Type: application/json<br/>
+
+  ```json
+  {
+    "data": {
+      "user_id": "SooJK37gzjIZfFNH0tlL",
+      "device_info": "t5k77PHnuDSrgEzvMJAj",
+      "authorization_status": "AUTHORIZED",
+      "access_token": "NLFSj7Kz30oHgolfIZtJ"
```

Comment:
if we are not storing the `accesstoken` , I don't think we can send it in this call?

Reply:
@sakshambhatt , @shreya-mishra ,

Reply:
we will store the accesstoken in the DB itself, the `PATCH` call will be authenticated, it was decided after discussion